### PR TITLE
Add ability for fetch --keep-keys and --drop-keys flags to do wildcard matching

### PR DESCRIPTION
**Pull request by tomreitz**

### CODE REVIEW

File: lightbeam/util.py [hunk 1]
```diff
@@ -40,4 +41,37 @@ def interpolate_params(params_structure, payload):
...
+# This function is used for both the endpoint selection in apply_filters() of api.py and
+# the keep-keys and drop-keys filtering in fetch.py
+def apply_selections(keys, keep, drop):
+    # `keep` and `drop` _should_ be arrays, but in case they're strings, we split them
+    if isinstance(keep, str): keep = keep.split(",")
+    if isinstance(drop, str): drop = drop.split(",")
+    # this will be the filtered set of keys
+    final_keys = []
+    # populate `final_keys` with `keys` that match `keep`
+    if len(keep)>0 and not (len(keep)==1 and keep[0]=="*"):
```
Comment (jayckaiser) [inline lightbeam/util.py:58]: Alternative: `if keep and keep != ["*"]:`

File: lightbeam/util.py [hunk 2]
```diff
@@ -40,4 +41,37 @@ def interpolate_params(params_structure, payload):
...
+        for payload_key, keep_key in list(itertools.product(keys, keep)):
+            if (keys_match(payload_key, keep_key)):
+                final_keys.append(payload_key)
+    else: final_keys = keys
+    # remove from `final_keys` keys that match `drop`
+    if len(drop)>0 and not (len(drop)==1 and drop[0]==""):
```
Comment (jayckaiser) [inline lightbeam/util.py:64]: Alternative: `if drop and drop != [""]:`

File: lightbeam/api.py [hunk 3]
```diff
@@ -53,67 +53,24 @@ def prepare(self):
...
+        # apply filters
+        my_endpoints = util.apply_selections(endpoints, self.lightbeam.selector, self.lightbeam.exclude)
         
         # make sure we have some endpoints to process
-        my_endpoints = list(set(selected_endpoints).difference(excluded_endpoints))
         if not my_endpoints:
             self.logger.critical("selector filtering left no endpoints to process; check your selector for typos?")
 
+        # make sure all selectors resolve to an endpoint
+        unknown_endpoints = list(set(my_endpoints).difference(endpoints))
```
Comment (jayckaiser) [inline lightbeam/api.py:64]: FYI, we don't need to convert this object to a list if we are only joining it with commas.

**Comment by tomreitz** [comment]
Thanks for the feedback, Jay! I've incorporated your suggestions.

**Review by jayckaiser — APPROVED** [review]

**Pull request merged**